extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,338 @@
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    /**
+     * Request new current conditions and forecast 10 days to the Weather Underground service
+     * and store the data in weatherData
+     *
+     * @return true if success or false in case of error
+     */
+    private boolean updateWeatherData() {
+        // Request new weather data to the Weather Underground service
+        WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+        weatherData = getWeatherData(USUAL_FEATURES, StringUtils.trimToEmpty(config.location));
+        return weatherData != null;
+    }
+
+    /**
+     * Request a geolookup feature to the Weather Underground service
+     * for a location given as parameter
+     *
+     * @param location the location to provide to the Weather Underground service
+     *
+     * @return true if success or false in case of error
+     */
+    private boolean checkWeatherLocation(String location) {
+        WeatherUndergroundJsonData data = getWeatherData(Collections.singleton(FEATURE_GEOLOOKUP), location);
+        return data != null;
+    }
+
+    /**
+     * Request new weather data to the Weather Underground service
+     *
+     * @param features the list of features to be requested
+     *
+     * @return the weather data object mapping the JSON response or null in case of error
+     */
+    private WeatherUndergroundJsonData getWeatherData(Set<String> features, String location) {
+        WeatherUndergroundJsonData result = null;
+        boolean resultOk = false;
+        String error = null;
+        String errorDetail = null;
+
+        try {
+
+            // Build a valid URL for the Weather Underground service using
+            // the requested features and the thing configuration settings
+            WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+
+            String urlStr = URL.replace("%APIKEY%", StringUtils.trimToEmpty(config.apikey));
+
+            String featuresStr = "";
+            for (String feature : features) {
+                featuresStr += "/" + feature;
+            }
+            urlStr = urlStr.replace("%FEATURES%", featuresStr.substring(1));
+
+            String lang = StringUtils.trimToEmpty(config.language);
+            if (lang.isEmpty()) {
+                urlStr = urlStr.replace("%SETTINGS%", "");
+            } else {
+                urlStr = urlStr.replace("%SETTINGS%", "lang:" + lang.toUpperCase());
+            }
+
+            urlStr = urlStr.replace("%QUERY%", location);
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from Weather Underground
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
```

Comment:
Keep in mind: It's the timeout for the whole request/response conversation. I'd rather set it too high than a little too low. The important point is to _have_ a timeout...

To be honest, I didn't spend too many thoughts for the concrete number, just consider it as an example and chose whatever sound reasonable to you.